### PR TITLE
Add tests for ndctl write-infoblock

### PR DESCRIPTION
**Pull request by harish-24**
Patch adds positive and negative tests using write infoblock and verifies it is works as expected.

Signed-off-by: Harish <harish@linux.ibm.com>

### CODE REVIEW

**Comment by harish-24** [comment]
Depends on https://github.com/avocado-framework/avocado/pull/3895

**Comment by harish-24** [comment]
```
 (1/2) memory/ndctl.py:NdctlTest.test_write_infoblock: PASS (39.57 s)
 (2/2) memory/ndctl.py:NdctlTest.test_write_infoblock_negate: FAIL: Enabling namespace must have failed (39.12 s)
RESULTS    : PASS 1 | ERROR 0 | FAIL 1 | SKIP 0 | WARN 0 | INTERRUPT 0 | CANCEL 0
JOB TIME   : 78.80 s
```
Without command support
```
 (1/2) memory/ndctl.py:NdctlTest.test_write_infoblock: CANCEL: Binary does not support write-infoblock (1.39 s)
 (2/2) memory/ndctl.py:NdctlTest.test_write_infoblock_negate: CANCEL: Binary does not support write-infoblock (1.36 s)
RESULTS    : PASS 0 | ERROR 0 | FAIL 0 | SKIP 0 | WARN 0 | INTERRUPT 0 | CANCEL 2
JOB TIME   : 2.98 s
```

File: memory/ndctl.py [hunk 1]
```diff
@@ -551,6 +562,83 @@ def test_region_capabilities(self):
...
+        self.write_infoblock(ns_name, self.get_size_alignval())
+        failed = False
+        try:
+            self.plib.enable_namespace(namespace=ns_name)
+        except pmem.PMemException:
+            self.log.error("Enabling namespace failed")
+            failed = True
+
+        if failed:
+            self.fail("Enabling namespace failed")
```
Comment (narasimhan-v) [inline memory/ndctl.py:603]:
```
try:
    self.plib.enable_namespace(namespace=ns_name)
except pmem.PMemException:
    self.log.error("Enabling namespace failed")
    self.fail("Enabling namespace failed")
```

**Comment by harish-24** [comment]
> Depends on [avocado-framework/avocado#3895](https://github.com/avocado-framework/avocado/pull/3895)

This can be considered for review as dependency is merged.

File: memory/ndctl.py [hunk 2]
```diff
@@ -551,6 +562,83 @@ def test_region_capabilities(self):
...
+            self.plib.enable_namespace(namespace=ns_name)
+        except pmem.PMemException:
+            self.log.info("Failed as expected")
+            failed = True
+        if not failed:
+            self.log.info(self.plib.run_ndctl_list())
+            self.fail("Enabling namespace must have failed")
+
+        idle_ns = self.plib.run_ndctl_list('-Ni -r %s' % region)
+        if len(idle_ns) > 1:
```
Comment (narasimhan-v) [inline memory/ndctl.py:630]: `if idle_ns:`

File: memory/ndctl.py [hunk 3]
```diff
@@ -551,6 +562,83 @@ def test_region_capabilities(self):
                     region=reg_name, mode='devdax', align=size)
                 self.plib.destroy_namespace(region=reg_name, force=True)
 
+    @avocado.fail_on(pmem.PMemException)
+    def write_infoblock(self, ns_name, align):
+        """
+        Write_infoblock on given namespace
+        """
+        if not self.plib.check_ndctl_subcmd("write-infoblock"):
+            self.cancel("Binary does not support write-infoblock")
```
Comment (narasimhan-v) [inline memory/ndctl.py:571]: do you want to cancel or fail here, considering we are in `test_X()` already ?

Comment (harish-24) [reply memory/ndctl.py:571]: `cancel` is more suitable considering most of the distro's ndctl package does not contain `write-infoblock` sub-command yet (fairly newer command). So it is kind of not supported with the distribution's ndctl package, and hence skipping here.

File: memory/ndctl.py (hunk 1 above)
Comment (harish-24) [reply memory/ndctl.py:603]: Thanks, this comment actually helped me find that even try-except is not required as `avocado.fail_on(pmem.PMemException)` takes care of it. Will make this change.

File: memory/ndctl.py (hunk 2 above)
Comment (harish-24) [reply memory/ndctl.py:630]: if `idle_ns` is true even for 1, but here condition is for more than 1.

Comment (narasimhan-v) [reply memory/ndctl.py:630]: Ah okay. I was thinking >= 1

File: memory/ndctl.py (hunk 1 above)
Comment (harish-24) [reply memory/ndctl.py]: @narasimhan-v Pushed with the changes, Thanks.

**Comment by narasimhan-v** [comment]
@harish-24 run log please

**Comment by harish-24** [comment]
[write_info.log](https://github.com/avocado-framework-tests/avocado-misc-tests/files/4764802/write_info.log)

**Pull request merged**